crates/bevy_ecs/src/system/commands/command.rs

```diff
@@ -116,9 +116,18 @@ pub fn insert_resource<R: Resource>(resource: R) -> impl Command {
 }
 
 /// A [`Command`] that removes a [`Resource`] from the world.
-pub fn remove_resource<R: Resource>() -> impl Command {
-    move |world: &mut World| {
-        world.remove_resource::<R>();
+///
+/// Returns an error if the resource does not exist.
+pub fn remove_resource<R: Resource>() -> impl Command<Result> {
+    move |world: &mut World| -> Result {
```

Comment:
This is too complicated and inconsistent: we should just be making `World::remove_resource` return a result, and then bubbling that up here.